gcloud-java-core/src/main/java/com/google/gcloud/AuthCredentials.java

```diff
@@ -243,6 +243,33 @@ public GoogleCredentials credentials() {
     }
   }
 
+  public static class NoAuthCredentials extends AuthCredentials {
```

Comment:
Nit and maybe for a separate PR: we could add some javadoc to our `XXXCredentials` classes